### PR TITLE
feat(module-context): mock entry point, BaseConfigBuilder refactor, docs restructure

### PR DESCRIPTION
**Pull request by odinr**
**Why is this change needed?**
`@equinor/fusion-framework-module-context` had no test double, so `FrameworkMockConfigurator.context` still hit the network in tests — the last core module without one. The configurator also predated the `BaseConfigBuilder` pattern used by `module-event`/`module-http`/`module-msal`, and its README had grown into a manual instead of an entry point.

**What is the current behavior?**
- Context has no `/mock` entry point; seeding a context in a test means faking a real `ContextClient` or navigation module.
- `ContextModuleConfigurator` predates `BaseConfigBuilder`; selector/parse-item files live loose at `src/` root.
- `FrameworkMockConfigurator.context` returns the real configurator — no test double, real I/O.
- The context module's README is a long manual; no `docs/` folder, unlike `module`/`module-http`.

**What is the new behavior?**
- New `@equinor/fusion-framework-module-context/mock` (`ContextMockConfigurator`, `enableContextMock`) — in-memory context backed by the real `ContextProvider` (validation/resolution/parent-propagation all still run for real).
- New `@equinor/fusion-framework-module-context/mock/fixtures` (`createContextItems`, `createContextItemFactory`) — realistic fixtures via an optional `@faker-js/faker` peer dependency, kept off the main `/mock` entry point.
- `ContextModuleConfigurator` now extends `BaseConfigBuilder`; `configurator.ts` → `ContextModuleConfigurator.ts`, split into `ContextModuleConfig.ts`/`ContextModuleConfigurator.interface.ts`; selectors moved to `src/selectors/`; `parseContextItem` moved to `src/utils/` and is now exported.
- `FrameworkMockConfigurator.context` now pins `contextMockModule` instead of the real module.
- `packages/modules/context/README.md` trimmed to an entry point; `docs/data-model.md`, `docs/lifecycle.md`, `docs/recipes.md` added and mirrored in vue-press with an updated sidebar.

**What is the intended behavior or invariant?**
The mock only substitutes the data source (`ContextClient`) — `validateContext`, `resolveContext`, and parent/child context propagation always run through the real `ContextProvider`/`ContextModuleConfigurator` logic, so a passing test against the mock reflects real resolution behavior, not a stubbed shortcut. `enableContextMock` never requires `@faker-js/faker`; only importing `/mock/fixtures` does.

**Does this PR introduce a breaking change?**
No public export names change on `@equinor/fusion-framework-module-context`. On `@equinor/fusion-framework`, `FrameworkMockConfigurator.context`'s return type changes from `ContextModuleConfigurator` to `ContextMockConfigurator` (which extends it) — code that only used documented configurator methods is unaffected; code relying on `.context` performing real I/O should switch to the new seeding methods.

**Impact assessment:**
- Breaking changes: No (see above)
- Version bump: `@equinor/fusion-framework-module-context` minor, `@equinor/fusion-framework` minor, `@equinor/fusion-framework-docs` patch (changesets included)
- Consumer impact: New opt-in `/mock` and `/mock/fixtures` subpaths; `FrameworkMockConfigurator.context` behavior improves (no more real I/O)
- Downstream impact: None outside these two packages

**Review guidance:**
Five commits, each independently reviewable: (1) `BaseConfigBuilder` refactor/file reorg, (2) the new mock/fixtures entry points, (3) docs restructure, (4) framework's `.context` accessor switch, (5) changesets.

**Additional context**
Full QA pass before pushing: Biome, `fusion-lint`, `tsc -b --force`, and the full Vitest suite (767 passing; one pre-existing unrelated failure in `react-router` from a missing `@equinor/fusion-imports` package). `pnpm build:docs` (vue-press) also builds cleanly.

**Related issues**
Closes equinor/fusion-core-tasks#1658

Note: this PR's base is `feat/fusion-testing`, not `main`, so the closing keyword won't auto-fire on merge here — it needs to also appear on the eventual `feat/fusion-testing` → `main` PR (none open yet) for GitHub to close it automatically.

### Checklist

- [x] Confirm completion of the self-review checklist
- [x] Confirm TSDoc captures intent for functions, hooks, components, classes, and named arrow functions
- [x] Confirm iterator blocks, decision gates, RxJS chains, and complex decisions explain why they exist
- [ ] Confirm React logic and derived values are resolved before markup when applicable — N/A, no React changes
- [x] Confirm README/docs are updated for user-facing changes
- [x] Confirm changes to target branch validation
  - Included files validated
  - No new linting warnings
  - Not a duplicate PR (checked existing)
- [x] Confirm adherence to code of conduct

### CODE REVIEW

File: packages/modules/context/src/ContextModuleConfigurator.ts [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+ * @remarks
+ * Extends {@link BaseConfigBuilder} and fully reuses its `_set`/`_buildConfig`
+ * machinery. Setters always pass an `async () => value` callback to `_set`,
+ * never the raw value directly — `_set` decides whether it received a value
+ * or a deferred callback by checking `typeof value_or_cb === 'function'`, so
+ * fields that are themselves functions (e.g. `validateContext`) would
+ * otherwise be misread as callbacks and invoked instead of stored.
+ */
+export class ContextModuleConfigurator
+  extends BaseConfigBuilder<ContextModuleConfig>
```
[the remark this answers is not in the document]
Comment (odinr) [reply packages/modules/context/src/ContextModuleConfigurator.ts:50]: Confirmed — verified against the pre-refactor `configurator.ts` (`public async createConfig(...)` returned `Promise<ContextModuleConfig>` directly) vs. `BaseConfigBuilder.createConfig` (`@sealed`, returns `Observable`). This matches the same trade-off `service-discovery`/`msal` accepted when they adopted `BaseConfigBuilder`, so I kept `createConfig`/`createConfigAsync` as-is rather than overriding the sealed method, and instead bumped the changeset to `major` with migration guidance to `createConfigAsync`. Fixed in 1a6b8ea05.

File: packages/modules/context/src/mock/ContextMockConfigurator.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    return this;
+  }
+
+  /**
+   * Escape hatch: overrides context resolution directly.
+   *
+   * @remarks
+   * For a custom `resolveContext` strategy or a shape the friendly seeding
+   * methods above did not cover — reaching for this means they were already
+   * tried and did not fit.
```
[the remark this answers is not in the document]
Comment (odinr) [reply packages/modules/context/src/mock/ContextMockConfigurator.ts]: Correct — `setResolver` only replaces `#resolve`, which backs `ContextClient.resolveContext(id)` (used by `setCurrentContextById` and the mock's initial-context lookup). `ContextProvider.resolveContext(item)` always goes through `relatedContexts`/`setRelatedContexts` instead, regardless of `setResolver`. Reworded the class-level "Escape hatch" bullet and the `setResolver` method doc to describe it as id-based lookup and call out that related-context resolution is unaffected. Fixed in 65c48cd61.

File: packages/framework/src/mock/FrameworkMockConfigurator.ts [hunk 3]
```diff
@@ -88,7 +89,7 @@ export class FrameworkMockConfigurator<
     this._pin(serviceDiscoveryMockModule);
     this._pin(httpMockModule);
     this._pin(servicesModule);
-    this._pin(contextModule);
+    this._pin(contextMockModule);
```
[the remark this answers is not in the document]
Comment (odinr) [reply packages/framework/src/mock/FrameworkMockConfigurator.ts:92]: Agreed, added a test in `mock-framework.test.ts` that seeds via `configurator.context.setCurrentContext(...)`, initializes the framework through `mockFramework`, and asserts `fusion.modules.context.currentContext` — mirroring the existing msal/serviceDiscovery coverage pattern and exercising the `_pin(contextMockModule)` path end-to-end. Fixed in 4cb793d58.

**Review by Noggling — APPROVED** [review]
